hibernate-core/src/main/java/org/hibernate/persister/entity/SingleTableEntityPersister.java

```diff
@@ -486,6 +486,11 @@ public String getTableName() {
 		return qualifiedTableNames[0];
 	}
 
+	@Override
+	public String[] getTableNames() {
+		return qualifiedTableNames;
```

Comment:
I've briefly checked the issue an maybe this approach could be used to solve that. Now the collection `referencedTables` is passed to procedure which joining parent tables in `JoinedSubclassEntityPersister`. It could be probably passed also to other places to optimize joining.